src/app/components/knob/knob.directive.js

```diff
@@ -53,33 +56,44 @@ class KnobController {
       }
     };
     angular.extend(knobOptions, this.options);
-    $element.knob(knobOptions);
+    this.$element.knob(knobOptions);
 
     // NOTE live update value
-    $scope.$watch(() => this.value,  (newVal, oldVal) => {
-      if (newVal !== oldVal) {
-        $timeout.cancel(this.timeoutHandler3);
-        this.timeoutHandler3 = this.updateValue();
+    $scope.$watch(() => this.value, (newVal, oldVal) => {
+      if (newVal !== oldVal || this.getCanvasValue() !== newVal) {
```

Comment:
keep this check simple, newVal !== oldVal and call this.getCanvasValue() inside updateValue()

Reply:
The suggested code would not work as updateValue() would never be called as the current condition contains a logical OR, not AND.

Reply:
the suggestion here is to don't spread the check in two functions, pass newVal, oldVal to getCanvasValue and give to getCanvasValue a proper name

Reply:
Well, now it makes sense :) . I'm always open to improvements, also in this case that basically is just an "Extract method".